controllers/osbuildconfig_controller.go

```diff
@@ -47,13 +53,54 @@ type OSBuildConfigReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *OSBuildConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling", "edgeDevice", req)
+
+	osBuildConfig, err := r.OSBuildConfigRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, err
+	}
 
-	// TODO(user): your logic here
+	if osBuildConfig.DeletionTimestamp != nil {
+		return ctrl.Result{}, nil
+	}
+
+	err = r.createOSBuildCR(ctx, osBuildConfig, logger)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *OSBuildConfigReconciler) createOSBuildCR(ctx context.Context, osBuildConfig *osbuilderprojectflottaiov1alpha1.OSBuildConfig, logger logr.Logger) error {
+	osBuildCurrentVersion := string(rune(*osBuildConfig.Status.LastVersion + 1))
+
+	osBuildConfigCp := osBuildConfig.DeepCopy()
```

Comment:
Do you really need to make a copy of the OSBuildConfig structure? If you do, then maybe just its `Spec.Details`

Reply:
You're right, thanks
Fixed it